mmedit/models/backbones/encoder_decoders/necks/aot_neck.py

```diff
@@ -0,0 +1,109 @@
+from mmcv.cnn import ConvModule
+
+from mmedit.models.registry import COMPONENTS
+
+
+@COMPONENTS.register_module()
+class AOTBlockNeck(nn.Module):
+    """Dilation backbone used in AOT-GAN model.
+    This implementation follows:
+    Aggregated Contextual Transformations for High-Resolution Image Inpainting
+    Args:
+        in_channels (int): Channel number of input feature.
+        dilation_rates (str): The dilation rates used
+        for AOT block. Default: 1+2+4+8.
+        num_aotblock (int): Number of AOT blocks. Default: 8.
+        act_cfg (dict): Config dict for activation layer, "relu" by default.
+        kwargs (keyword arguments).
+    """
+
+    def __init__(self,
+                 in_channels=256,
+                 dilation_rates='1+2+4+8',
+                 num_aotblock=8,
+                 act_cfg=dict(type='ReLU'),
+                 **kwargs):
+        super().__init__()
+
+        self.dilation_rates = list(map(int, list(dilation_rates.split('+'))))
+        self.num_aotblock = num_aotblock
+
+        self.model = nn.Sequential()
+
+        for i in range(self.num_aotblock):
+            self.model.add_module(
+                f'aotblock{i + 1}',
+                AOTBlock(
+                    in_channels=in_channels,
+                    dilation_rates=self.dilation_rates,
+                    act_cfg=act_cfg,
+                ))
+
+    def forward(self, x):
+        return self.model(x)
+
+
+class AOTBlock(nn.Module):
+    """AOT Block which constitutes the dilation backbone.
+    This implementation follows:
+    Aggregated Contextual Transformations for High-Resolution Image Inpainting
+    The AOT Block adopts the split-transformation-merge strategy:
+    Splitting: A kernel with 256 output channels is split into four
+               64-channel sub-kernels.
+    Transforming: Each sub-kernel performs a different transformation with
+                  a different dilation rate.
+    Splitting: Sub-kernels with different receptive fields are merged.
+    Args:
+        in_channels (int): Channel number of input feature.
+        dilation_rates (str): The dilation rates used for AOT block.
+        act_cfg (dict): Config dict for activation layer, "relu" by default.
+        kwargs (keyword arguments).
+    """
+
+    def __init__(self,
+                 in_channels,
+                 dilation_rates,
+                 act_cfg=dict(type='ReLU'),
+                 **kwargs):
+        super().__init__()
+        self.dilation_rates = dilation_rates
+
+        for i, dilation_rate in enumerate(dilation_rates):
+            self.__setattr__(
+                f'block{i + 1}',
+                nn.Sequential(
+                    nn.ReflectionPad2d(dilation_rate),
+                    ConvModule(
+                        in_channels,
+                        in_channels // 4,
+                        kernel_size=3,
+                        dilation=dilation_rate,
+                        act_cfg=act_cfg)))
```

Comment:
Similar to above, if the name `block{i + 1}` is not necessary, we can use `nn.ModuleLIst` instead.